File: core/src/net/jorhlok/multiav/SEffect.java
package net.jorhlok.multiav;

import com.badlogic.gdx.Gdx;
import com.badlogic.gdx.audio.Sound;

/**
 *
 * @author joshm
 */
public class SEffect {
    //setup
    protected String Name;
    protected String URI;
    protected float GlobalVolume = 1;
    
    //runtime
    protected Sound Data;
    
    public SEffect(String key, String uri) {
        Name = key;
//        Data = Gdx.audio.newSound(new FileHandle(uri));
        URI = uri;
    }
    
    public void Generate() {
        Data = Gdx.audio.newSound(Gdx.files.internal(URI));
    }
    
    public String getName() {
        return Name;
    }
    
    public void setVolume(float v) {
        GlobalVolume = Math.max(v, 0);
    }
    
    public long play() {
        return play(1,1,0,false);
    }
    public long play(boolean loop) {
        return play(1,1,0,loop);
    }
    
    public long play(float volume) {
        return play(volume,1,0,false);
    }
    
    public long play(float volume, boolean loop) {
        return play(volume,1,0,loop);
    }
    
    public long play(float volume, float speed) {
        return play(volume,speed,0,false);
    }
    
    public long play(float volume, float speed, boolean loop) {
        return play(volume,speed,0,loop);
    }
    
    public long play(float volume, float speed, float pan) {
        return play(volume,speed,pan,false);
    }

    public long play(float volume, float speed, float pan, boolean loop) {
        try {
            long id = Data.play(volume*GlobalVolume, speed, pan);
            Data.setLooping(id, loop);
            return id;
        } catch (Exception e) {
            return -1;
        }
    }
    
    public void stop() {
        try {
            Data.stop();
        } catch (Exception e) {
            //nothing
        }
    }
    
    public void stop(long id) {
        try {
            Data.stop(id);
        } catch (Exception e) {
            //nothing
        }
    }
    
    public void dispose() {
        try {
            Data.dispose();
        } catch (Exception e) {
            //nothing
        }
    }
}

File: core/src/net/jorhlok/multiav/SEffect.kt
package net.jorhlok.multiav

import com.badlogic.gdx.Gdx
import com.badlogic.gdx.audio.Sound

/**

 * @author joshm
 */
class SEffect(key: String, protected var URI: String) {
    //setup
    var name: String
        protected set
    protected var GlobalVolume = 1f

    //runtime
    protected var Data: Sound? = null

    init {
        name = key
    }

    fun Generate() {
        Data = Gdx.audio.newSound(Gdx.files.internal(URI))
    }

    fun setVolume(v: Float) {
        GlobalVolume = Math.max(v, 0f)
    }

    fun play(loop: Boolean): Long {
        return play(1f, 1f, 0f, loop)
    }

    fun play(volume: Float, loop: Boolean): Long {
        return play(volume, 1f, 0f, loop)
    }

    fun play(volume: Float, speed: Float, loop: Boolean): Long {
        return play(volume, speed, 0f, loop)
    }

    @JvmOverloads fun play(volume: Float = 1f, speed: Float = 1f, pan: Float = 0f, loop: Boolean = false): Long {
        try {
            val id = Data!!.play(volume * GlobalVolume, speed, pan)
            Data?.setLooping(id, loop)
            return id
        } catch (e: Exception) {
            return -1
        }

    }

    fun stop() {
        try {
            Data?.stop()
        } catch (e: Exception) {
            //nothing
        }

    }

    fun stop(id: Long) {
        try {
            Data?.stop(id)
        } catch (e: Exception) {
            //nothing
        }

    }

    fun dispose() {
        try {
            Data?.dispose()
        } catch (e: Exception) {
            //nothing
        }

    }
}
